src/megaapi_impl.cpp

```diff
@@ -21503,82 +21399,42 @@ void MegaApiImpl::sendPendingRequests()
             const char *name = request->getName();
 
             auto nextSyncTag = client->nextSyncTag();
-            MegaSyncPrivate *sync = new MegaSyncPrivate(localPath, name, node->nodehandle, nextSyncTag);
-            sync->setRegExp(request->getRegExp());
-
-            SyncError syncError = NO_SYNC_ERROR;
 
             std::unique_ptr<char []> remotePath{getNodePathByNodeHandle(request->getNodeHandle())};
             if (!remotePath)
             {
                 e = API_ENOENT;
-                syncError = REMOTE_NODE_NOT_FOUND;
-                request->setNumDetails(syncError);
+                request->setNumDetails(REMOTE_NODE_NOT_FOUND);
                 break;
             }
 
             SyncConfig syncConfig{nextSyncTag, localPath, name, request->getNodeHandle(), remotePath.get(),
                                   0, regExpToVector(request->getRegExp())};
 
-            e = client->addsync(syncConfig, DEBRISFOLDER, NULL, syncError, true, sync);
-            request->setNumDetails(syncError);
-            if (!e || !isSyncErrorPermanent(syncError))
+            UnifiedSync* unifiedSync = nullptr;
+            e = client->addsync(syncConfig, DEBRISFOLDER, NULL, true, unifiedSync, false);  // notifyApp = false since so we don't notify until after fireOnSyncAdded
+            request->setNumDetails(syncConfig.getError());
+            if (unifiedSync)
```

Comment:
Testing unifiedSync checks if the syncConfig was added to our set or not (and if it was then we should notify the app with fireOnSyncAdded() )  - !e isn't quite the same because some types of error may mean we didn't even add it, but other types of error mean we added it to the set of syncConfigs, but were not able to start it - or there may even be some errors (now or in future) where we can't differentiate those just on error code.   We could split those two cases into separate functions - effectively separating the checkSyncConfig(), and the remainder - though we already discussed and decided not to do that a while back, I'm happy to revisit that though.

Reply:
Ouch! I didn't noticed about the reassignment of the `error` upon `enableSync()` here:
``` cpp
    if (!e)
    {
        // if we got this far, the syncConfig is kept (in db and in memory)
        unifiedSync = syncs.appendNewSync(config, *this);

        e = unifiedSync->enableSync(false, notifyApp);

        syncactivity = true;
        config = unifiedSync->mConfig;  // so the caller can easily check the config they passed in
    }
```
My bad, sorry. You're very correct. Let's keep it as it is right now. Thanks.